src/schemathesis/extra/pytest_plugin.py

```diff
@@ -90,13 +90,24 @@ def _parametrize(
 
     def _make_test(self, endpoint: Endpoint) -> Callable:
         try:
-            return create_test(endpoint, self.test_function)
+            return create_test(endpoint, self.test_function, stateful=self.schemathesis_case.stateful)
         except InvalidSchema:
             return lambda: pytest.fail("Invalid schema for endpoint")
 
     def collect(self) -> List[Function]:  # type: ignore
         """Generate different test items for all endpoints available in the given schema."""
-        try:
+        try:  # pylint: disable=too-many-nested-blocks
+            if self.schemathesis_case.stateful:
+                items = []
+                for endpoint in self.schemathesis_case.get_all_endpoints():
+                    for dependencies in endpoint.dependencies.values():
+                        if dependencies:
+                            for dependent_endpoint in self.schemathesis_case.sort_by_requirements(dependencies):
+                                for item in self._gen_items(dependent_endpoint):
+                                    items.append(item)
+                    for item in self._gen_items(endpoint):
```

Comment:
The current approach goes another way around - after each endpoint test, its dependencies are checked and executed if needed, and then it goes recursively (it is implemented only inside the runner, though). This one adds dependencies first and then the endpoint itself. It feels like in both cases results are reused and therefore they can hit more lines of the application under test and they both feel equally capable, but I am not completely sure about it - will do some experiments (however with this one the number of tests is known before the execution)

Reply:
yeah, the idea was to execute tests against dependencies to gather required values (and possibly modify the same resources) and then execute tests for the (target) endpoint.  You are right, results are reused. Correct me if I'm wrong, but I think your recursive approach might test even deeper state-space thanks to testing transitive dependencies. As you say, some experiments are needed, also for different recursion depth limits.